method/link.js

```diff
@@ -134,6 +134,53 @@ module.exports = function Link (ssb) {
         }),
         pull.collect(cb)
       )
+    },
+    findGroupBySubgroupId (groupId, cb) {
+      if (!isGroup(groupId)) return cb(new Error(`findSubgroupByGroupId expected a groupId, got ${groupId} instead.`))
+
+      const query = [{
+        $filter: {
+          value: {
+            content: {
+              type: 'link/group-subgroup',
+              child: groupId,
+              tangles: {
+                link: { root: null, previous: null }
+              }
+            }
+          }
+        }
+      }]
+
+      pull(
+        // NOTE: using ssb-query instead of ssb-backlinks
+        // because the backlinks query will get EVERY message which contains the groupId in it, which will be a LOT for a group
+        // then filters that massive amount down to the ones which have the dest in the right place
+        ssb.query.read({ query }),
+        pull.unique(link => link.value.content.parent),
+        pull.filter(groupSubgroupLink.spec.isRoot),
+        pull.filter(link => {
+          return link.value.content.parent === link.value.content.recps[0]
+          // it would be very strange for a link to be created like this
+          // but we should consider it unsafe and ignore it I think
+        }),
+        pull.map(link => {
+          const { parent, recps } = link.value.content
+
+          return {
+            linkId: link.key,
+            groupId,
+            parentGroupId: parent,
+            recps,
+            states: [{
+              head: link.key,
```

Comment:
is this `head` or `key` on all general `crut` things... I can't remember

Reply:
I cant remember either, im just using patterns from above